src/main/java/com/google/api/codegen/transformer/InitCodeTransformer.java

```diff
@@ -599,7 +614,8 @@ private InitValueView getInitValue(GapicMethodContext context, InitCodeNode item
                 .variantType(namer.getOneofVariantTypeName(item.getOneofConfig()))
                 .build());
       }
-
+      fieldSetting.required(
+          hasFieldConfig(context.getMethodConfig().getRequiredFieldConfigs(), fieldConfig));
```

Comment:
This looks like it has O(n^2) runtime characteristics - would that be concerning?

Reply:
Depends... n in this case is almost certainly < 10. So probably not a big deal. If it's concerning to you, the alternative is to initialize a `Map<MethodConfig, Set<RequiredFieldConfig>>` or something in the parent scope.

Lmk.

Reply:
Actually I'll just change it.

Reply:
Changed my mind, I tried using a `Set`, but it doesn't work. The hash codes of the `FieldConfig` derived from `InitCodeNode` is different from the ones derived from `GapicMethodContext`. The equals comparison does work, however.

Reply:
Can we do comparison on the field name instead of the entirety of the field config?

Reply:
Done